### PR TITLE
Skip MCFakeFile parentage

### LINKED ISSUE #8337: Do not put MCFakeFile as parent file when ACDC is created.
Is it necessary to MCFakeFile as parents since it won't be added in DBS anyway.

### PR DESCRIPTION
**Pull request by amaltaro**
Fixes #8337
Also contains changes from #8338

@ticoann I think we'll need to have this into the new agent and also backported to the current production agents. I'm going to run a quick ACDC test, just to check whether it doesn't break anything.

### CODE REVIEW

**Comment by ticoann** [comment]
patch looks good to me. But we also need the patch fore ErrorHandling part so we wan't put the MCFakefile initially in the ACDC

**Comment by amaltaro** [comment]
Good point Seangchan. I updated the patch. I'm also not sure whether we could have a third case where parents would not be set at all. Can you review it again please?

**Comment by amaltaro** [comment]
If everything is fine, I'll patch vocms0253 some point tomorrow.

**Comment by ticoann** [comment]
Thanks Alan patch looks good to me.

**Comment by ticoann** [comment]
I am just wondering if this workflow failed so many jobs in the first step why would ops create ACDC instead of cloning?

**Comment by amaltaro** [comment]
It wasn't that bad, it had 7k failures against 100k success jobs. Now, don't ask me where that 47k offset comes from :)

**Comment by amaltaro** [comment]
The component took 10h to go through those wmbs operations. I suggest we push these changes in in the new agent only (and properly validate it as well).

**Comment by ticoann** [comment]
If that agent went through this, yes I agree we put this on only new agents.

**Comment by amaltaro** [comment]
Trying to figure why parents are no longer put in the acdc fileset docs... I can't see any issues with this patch nor with the one already in

File: src/python/WMCore/ACDC/DataCollectionService.py [hunk 1]
```diff
@@ -97,10 +97,14 @@ def failedJobs(self, failedJobs, useMask=True):
             for fInfo in inputFiles:
                 if fInfo["merged"]:
```
Comment (amaltaro) [inline src/python/WMCore/ACDC/DataCollectionService.py]: Haaa... it was this merged value that is a string, either '0' or '1', so it evaluates always to True :(

**Comment by amaltaro** [comment]
@ticoann it has been backported already to 1.1.8, but please review even if it's already merged :)

**Pull request merged**